### PR TITLE
Re-Former

### PR DESCRIPTION
**Pull request by sumon0002001**
I implement migrations and models to complete the task.

## Built With

- Ruby 2.7.0
- Ruby on Rails 6.0.3.4
- Visual Studio Code

## Getting Started
To get a local copy up and running follow these simple example steps.

- You need Ruby 2.7.0 and rails >= 6.0.3.4
- Open a terminal or command prompt interface on your PC.
- Clone the repo with: git clone https://github.com/sumon0002001/re-former
- On the terminal navigate to the project directory using cd re-former
- run bundle install
- rails db:migrate

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    if @user.update(user_params)
+      redirect_to edit_user_path
+    else
+      render :edit
+    end
+  end
+
+  def user_params
+    params.require(:user).permit(:username, :password, :email)
+  end
```
Comment (kevcap) [inline app/controllers/users_controller.rb:38]: Please use the private method for user_params

**Review by kevcap — CHANGES_REQUESTED** [review]
# [STATUS] CHANGES REQUIRED♻️

#### Hey @sumon0002001,  your project is looking really good, however, to accomplish as the actual project, you required to do these changes, listed below:
 
## [REQUIRED CHANGES]
- [x] Please use the private `user_params` method
- [x] Currently, your page isn't redirecting us to new_user_path, if the user was created.
- [x] Please set your users/new as the root of your project.
- [x] Please use form_authenticity_token at your view
```html
<form action="users_path" method="post" accept-charset="UTF-8">
  <label for="user[username]">Username</label>
  <input name="user[username]" type="text" />
  <input
    type="hidden"
    name="authenticity_token"
    value="<%= form_authenticity_token %>"
  />
  <input type="submit" value="Submit" />
</form>
```
([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form1.png) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))


## [OPTIONAL CHANGES]
- [ ] partial created for user form and reused in edit and new view
- [ ] form view to displays a list of the error messages (with #errors and full_messages usage) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_extra_shared_error.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
- [ ] validate the email format by using either a regular expression or a built in URI method.

## Please make the required changes and resubmit for a new code review once you are ready💪

## As always, I wish you all the very best and see you both at the top🚀


👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> gmail</a> [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

<sub>*We're here to help you, to ensure that you're using the best practices and getting you ready for the job market. In the real world, you'll have a lot of change requests for your real clients, it's good to get used to it.</sub>

**Review by Temmarie — CHANGES_REQUESTED** [review]
## Changes Required! :recycle:

### Hi @sumon0002001  ! :wave: :wave:

#### You have done well so far :clap: However, you did not finish all the required changes given by the previous TSE. Kindly work on them and submit them for another code review.


**Recommended**

- [ ] I can't create a new user as the page isn't redirecting us to new_user_path if the user was created.

_**Remember to use the correct Gitflow in your next project.**_

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink:
__Grace Tamara Ekunola __
[GitHub](https://github.com/temmarie) | [LinkedIn](https://www.linkedin.com/in/ekunola-grace/) | [Twitter](https://twitter.com/TemmarieW)

**Review by Temmarie — APPROVED** [review]
## Status: Approved :heavy_check_mark:

### Hi @sumon0002001 ! 👋 👋 

#### Thank you for making all of the required changes, your project is approved :clap: :clap: :clap: :tada:

### You may merge, good luck with your next projects! :rocket:
### Happy Coding! :+1:
__Grace Tamara Ekunola __
[GitHub](https://github.com/temmarie) | [LinkedIn](https://www.linkedin.com/in/ekunola-grace/) | [Twitter](https://twitter.com/TemmarieW)

**Pull request merged**